File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
from collections import deque
def solution(s):
    answer = 0
    s=deque(s)  
    stack=[]
    open=["[","(","{"]
    close=["]",")","}"]
    for j in range(0,len(s)):
        #print(s)
        stack=[]
        a=True
        for i in s:
            if i in open:
                stack.append(i)
            elif i in close :
                if stack and stack[-1] == open[close.index(i)]:
                    stack.pop()
                else:
                    a=False
                    break
        if stack:
            a=False
       
        if a:
            answer+=1
        s.append(s.popleft())
        print(a)
                
            
    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = 0; 
        for(int i = 0 ;i<s.length();i++){
            // System.out.println(s);
            if(perfectClose(s)){
                answer++;
            }
            
            s = s.substring(1).concat(s.substring(0,1));
        }
        
        return answer;
    }
    public boolean perfectClose(String s){
        
        String open = "({[";
        String close = ")}]";
        Stack<String> stack = new Stack<>();
        
        for(int i = 0 ;i<s.length();i++){
            if(open.contains(""+s.charAt(i))){
                  stack.push(""+s.charAt(i));
            }
            else{
                if (stack.isEmpty()){
                    return false;
                }
                if(open.indexOf(stack.pop()) != close.indexOf(""+s.charAt(i))){
                    return false;
                }
            }
          
                     
         }
       
        if(stack.isEmpty()){
            return true;
        }else{
            return false;
        }
        
    }
}